engine.py

```diff
@@ -46,6 +48,34 @@ def update_fov(self) -> None:
         # If a tile is "visible" it should be added to "explored".
         self.game_map.explored |= self.game_map.visible
 
+    def update_aoe(
```

Comment:
I don't think this needs to be a method of Engine or GameMap.  This might be moved inline to AreaRangedAttackHandler.